tools/cldr-apps/src/main/java/org/unicode/cldr/web/SurveyAjax.java

```diff
@@ -3185,47 +3173,177 @@ public String getValue() {
+            out.write("var org = '" + myUser.org + "';\n");
+            out.write("var surveyUserPerms = {\n");
+            out.write("  userExist: (surveyUser != null),\n");
+            out.write("  userCanImportOldVotes: " + myUser.canImportOldVotes() + ",\n");
+            out.write("  userCanUseVettingSummary: " + UserRegistry.userCanUseVettingSummary(myUser) + ",\n");
+            out.write("  userCanMonitorForum: " + UserRegistry.userCanMonitorForum(myUser) + ",\n");
+            out.write("  userIsTC: " + UserRegistry.userIsTC(myUser) + ",\n");
+            boolean userIsVetter = !UserRegistry.userIsTC(myUser) && UserRegistry.userIsVetter(myUser);
+            out.write("  userIsVetter: " + userIsVetter + ",\n");
+            out.write("  userIsLocked: " + UserRegistry.userIsLocked(myUser) + ",\n");
+            out.write("  hasDataSource: " + curSurveyMain.dbUtils.hasDataSource() + ",\n");
+            out.write("};\n");
+
+            out.write("var surveyUserURL = {\n");
+            out.write("  myAccountSetting: 'survey?do=listu',\n");
+            out.write("  disableMyAccount: \"lock.jsp?email='+userEmail\"+userEmail,\n");
+            out.write("  recentActivity: \"myvotes.jsp?user=\"+userID+\"&s=\"+surveySessionId,\n");
+            out.write("  xmlUpload: \"upload.jsp?a=/cldr-apps/survey&s=\"+surveySessionId,\n");
+            out.write("  manageUser: \"survey?do=list\",\n");
+            out.write("  flag: \"tc-flagged.jsp?s=\"+surveySessionId,\n");
+            out.write("  about: \"about.jsp\",\n");
+            out.write("  browse: \"browse.jsp\",\n");
+            out.write("};\n");
+
+            if (UserRegistry.userIsAdmin(myUser)) {
+                out.write("surveyUserURL.adminPanel = 'survey?dump=" + SurveyMain.vap + "';\n");
+            }
+        } else {
+            // User session not present. Set a few things so that we don't fail.
+            out.write("var surveyUser = null;\n");
+            out.write("var surveyUserURL = {};\n");
+            out.write("var surveyUser = null;\n");
+            out.write("var organizationName = null;\n");
+            out.write("var org = null;\n");
+            out.write("var surveyUserPerms = {userExist: false,};\n");
+        }
+        out.write("var warnIcon = \"" + WebContext.iconHtml(request, "warn", "Test Warning") + "\";\n");
+        out.write("var stopIcon = \"" + WebContext.iconHtml(request, "stop", "Test Error") + "\";\n");
+        out.write("var WHAT_GETROW = '" + SurveyAjax.WHAT_GETROW + "';\n");
+        out.write("var WHAT_SUBMIT = '" + SurveyAjax.WHAT_SUBMIT + "';\n");
+        out.write("var TARGET_DOCS = '" + WebContext.TARGET_DOCS + "';\n");
+        out.write("var TRANS_HINT_LOCALE = '" + SurveyMain.TRANS_HINT_LOCALE + "';\n");
+        out.write("var TRANS_HINT_LANGUAGE_NAME = '" + SurveyMain.TRANS_HINT_LANGUAGE_NAME + "';\n");
+        out.write("var WHAT_GETROW = '" + SurveyAjax.WHAT_GETROW + "';\n");
```

Comment:
Oops WHAT_GETROW is accidentally duplicated here. It's harmless but I'll remove it in the next PR.

Reply:
it's somewhat temporary anyway… 

Reply:
these WHAT_ things shouldn't need to be communicated to the client… hard coded would be fine. But no reason to change it now.